core/src/edu/asu/c3simulator/widgets/NavigationPanel.java

```diff
@@ -0,0 +1,346 @@
+package edu.asu.c3simulator.widgets;
+
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import com.badlogic.gdx.Game;
+import com.badlogic.gdx.Screen;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+
+/**
+ * This class creates a navigation panel by creating the buttons and linking them with
+ * their screens. It also supports having one level of sub-buttons to a parent button. A
+ * parent button will only show and hide sub-buttons when clicked. Only one of the parent
+ * buttons can show its sub-buttons at the same time. If a parent button is clicked, the
+ * already extended sub-buttons will collapse before extending the new set of sub-buttons.
+ * 
+ * @author Alyahya, Mohammed
+ */
+public class NavigationPanel extends Table
+{
+	private class NavigationButtonListener extends ClickListener
+	{
+		String screenName;
+		Screen targetedScreen;
+		
+		public NavigationButtonListener(String screenName, Screen targetedScreen)
+		{
+			this.screenName = screenName;
+			this.targetedScreen = targetedScreen;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			System.out.println(screenName);
+			game.setScreen(targetedScreen);
+		}
+	}
+	
+	private class CollapsibleButtonListener extends ClickListener
+	{
+		String screenName;
+		
+		public CollapsibleButtonListener(String screenName)
+		{
+			this.screenName = screenName;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			if (displayedCollapsibleButton.equalsIgnoreCase(screenName))
+				displayedCollapsibleButton = "";
+			else
+				displayedCollapsibleButton = screenName;
+			recreateNavigationPanel();
+		}
+	}
+	
+	private Game game;
+	private Skin skin;
+	private LinkedHashMap<String, Screen> buttons;
+	private LinkedHashMap<String, LinkedHashMap<String, Screen>> subButtons;
```

Comment:
1: Ah. It sounds like I misinterpreted, and encapsulation still stands. Good.

2: Noted.

3: I understand we are not talking about the end-user. My refactoring goal is this: maintain the functionality promised by the class through all public data members and functions, while simplifying the encapsulated details, namely by getting rid of the nested maps, and making both listeners static, if possible. 
3: With that said, I see 2 main objectives that are achieved by the private portions of this class: firstly, it keeps track of the order and relationship between buttons; secondly, it updates the panel in real time, to react to changes in the order and relationship. Listeners can accomplish half of that with ease; the other half can be accomplished by keeping a map to track the listeners, which will function almost as nodes. Alternatively, the map can be discarded, and instead defer to Table's actor tracking, although this would be much less clean. (I am partial to the former option)

4: I saw that. This widget has a lot of good features and functionality. I'm going to go ahead and refactor it a bit, using all of this as a solid foundation. I should have some code for you to look at later this weekend.